src/Common/src/Interop/Unix/libcurl/Interop.SafeCurlHandle.cs

```diff
@@ -37,6 +40,19 @@ protected override bool ReleaseHandle()
 
         internal sealed class SafeCurlMultiHandle : SafeHandle
         {
+            private readonly CancellationTokenSource _cancellationTokenSource = new CancellationTokenSource();
```

Comment:
It looks like we only ever use _cancellationTokenSource.CancellationToken.IsCancellationRequested, and never actually register any callbacks with the token.  In that case, it'd be better to just make this a volatile bool that gets checked and transitioned from false to true when necessary, since that's all IsCancellationRequested is doing in this case, and by just using a bool, we avoid the CTS allocation per multi handle.